File: python/statusbar.py
#
# statusbar.py
#
#     In this class the statusbar as used in Matlab, LabView and now Python is created
#
#  modifications
#     15-dec-2023   JM  initial version

from PySide6.QtWidgets import QStatusBar

from datetime import datetime

class statusBarNew(QStatusBar) :
  
  # constructor

  def __init__(self) :
    super().__init__()

  # setText
  #
  #     this method shows the text for the number of seconds
    
  def setText(self,txtToDisplay,timeToDisplay) :

    currentTime = datetime.now().strftime("%H:%M:%S")
    txtToDisplay = currentTime + " >>> " + txtToDisplay
    self.showMessage(txtToDisplay,(timeToDisplay * 1000))

File: c-sources/statusbar/statusbar.cpp
//
// statusbar.cpp
//
//    implementation for the <statusBar> class as used in LabView and Matlab
//
//  modifications
//    12-jan-2024   JM  initial version

#include "statusbar.h"
#include <QString>
#include <Qtime>

#include <stdio.h>

// constructore

statusBarNew::statusBarNew() {
}

// setText
//
//      displays <txtToDisplay> for <timeToDisplay> seconds in the statusbar. The message is 
//      prefixed with the current time

void statusBarNew::setText(QString txtToDisplay, float timeToDisplay) {

  // get the current time in a format hh:mm:ss

  QTime time = QTime::currentTime();
  QString formattedTime = time.toString("hh:mm:ss");
  
  // and display
  
  txtToDisplay = formattedTime + QString(" >>> ") + txtToDisplay;
  showMessage(txtToDisplay,int(timeToDisplay * 1000));
}

    